packages/resource-deployment/scripts/delete-outdated-pools.sh

```diff
@@ -0,0 +1,149 @@
+"
+    exit 1
+}
+
+deletePool() {
+    poolId=$1
+
+    echo "deleting pool $poolId"
+    az batch pool delete --account-name $batchAccountName --pool-id $poolId --yes
+
+    waitForDelete $poolId
+    echo "done"
+}
+
+waitForDelete() {
+    poolId=$1
+
+    checkIfPoolExists $poolId
+    waiting=false
+    end=$((SECONDS + $deleteTimeout))
+    while [ "$poolExists" == "true" ] && [ $SECONDS -le $end ]; do
+        if [ "$waiting" != true ]; then
+            waiting=true
+            echo "Waiting for $poolId to delete"
+            printf " - Running .."
+        fi
+
+        sleep 5
+        printf "."
+        checkIfPoolExists
+    done
+
+    if [ "$poolExists" == "true"]; then
+        echo "Pool did not finish deleting within $deleteTimeout seconds"
+    fi
+}
+
+checkIfPoolExists() {
+    poolId=$1
+
+    poolExists=$(az batch pool list --account-name "$batchAccountName" --query "[?id=='$poolId']" -o tsv)
+    if [[ -z $poolExists ]]; then
+        poolExists=false
+    else
+        poolExists=true
+    fi
+}
+
+deletePoolWhenNodesAreIdle() {
+    poolId=$1
+
+    command="deletePool $poolId"
+    commandName="Delete pool $poolId"
+    . "${0%/*}/run-command-when-batch-nodes-are-idle.sh"
+}
+
+compareConfigFileToDeployedConfig() {
+    poolId=$1
+    batchConfigPropertyName=$2
+    templateFileParameterName=$3
+
+    query="[?id=='$poolId'].$batchConfigPropertyName"
+
+    expectedValue=$(cat $parameterFilePath | jq -r ".parameters.$templateFileParameterName.value")
+    actualValue=$(az batch pool list --account-name "$batchAccountName" --query "$query" -o tsv)
+
+    if [ "$expectedValue" != "$actualValue" ]; then
+        echo "$batchConfigPropertyName for $poolId must be updated from $actualValue to $expectedValue."
+        shouldDeletePool=true
+    fi
+}
+
+deletePoolIfOutdated() {
+    poolId=$1
+    poolPropertyNamePrefix=$2
+
+    shouldDeletePool=false
+
+    checkIfPoolExists $poolId
+    if [ "$poolExists" == "false" ]; then
+        echo "Pool $poolId not yet created."
+        return
+    fi
+
+    compareConfigFileToDeployedConfig $poolId "vmSize" "${poolPropertyNamePrefix}VmSize"
+    if [ $shouldDeletePool == "true" ]; then
+        deletePool $poolId
+        return
+    fi
+
+    compareConfigFileToDeployedConfig $poolId "maxTasksPerNode" "${poolPropertyNamePrefix}MaxTasksPerNode"
+    if [ $shouldDeletePool == "true" ]; then
+        deletePoolWhenNodesAreIdle $poolId
+        return
+    fi
+
+    echo "$poolId does not need to be recreated."
+}
+
+# Read script arguments
+while getopts ":b:p:t:r:" option; do
+    case $option in
+    b) batchAccountName=${OPTARG} ;;
+    p) parameterFilePath=${OPTARG} ;;
+    t) deleteTimeout=${OPTARG} ;;
+    r) resourceGroupName=${OPTARG} ;;
+    *) exitWithUsageInfo ;;
+    esac
+done
+
+# Print script usage help
+if [[ -z $batchAccountName ]] || [[ -z $parameterFilePath ]] || [[ -z $resourceGroupName ]]; then
```

Comment:
don't ask for inputs that we can figure out ourselves in code. call get-resource-names.sh to set the batchAccountName